src/Language/JavaScript/Dependencies.hs

```diff
@@ -0,0 +1,379 @@
+module Language.JavaScript.Dependencies
+  ( JSAST
+  , Dependencies
+  , jsAstDependencies
+  ) where
+
+import PacchettiBotti.Prelude
+
+import qualified Data.Char as Char
+import qualified Data.List as List
+import qualified Data.Set as Set
+import qualified Data.Text as Text
+import qualified Network.URI as URI
+import qualified Turtle
+
+import Data.Set (Set)
+import Language.JavaScript.Parser.AST
+
+type ModuleId = String
+type Dependencies = Set Text
+
+dependency :: ModuleId -> Dependencies
+dependency moduleId
+  | not . Turtle.absolute $ Turtle.decodeString moduleId
+  , not $  "." `List.isPrefixOf` moduleId
+  = foldMap Set.singleton $ case Text.split (== '/') $ Text.pack moduleId of
+      scope : name : _
+        | "@" `Text.isPrefixOf` scope
+        -> Just $ scope <> "/" <> name
+      name : _ -> Just name
+      _ -> Nothing
+  | otherwise
+  = mempty
+
+isCommonJsRequire :: JSExpression -> Bool
+isCommonJsRequire = \case
+  JSMemberDot (JSIdentifier _ "module") _ (JSIdentifier _ "require") -> True
+  JSCallExpressionDot (JSIdentifier _ "module") _ (JSIdentifier _ "require") -> True
+  JSIdentifier _ "require" -> True
+  _ -> False
+
```

Comment:
I guess in general this module looks pretty mechanical, but docstrings (or even just a docstring for the whole module) could help understand what's going on

Reply:
I added a comment to `jsAstDependencies`.